### ISSUE TITLE
TypeLoadException when validating a target that contains a Uri property

### ISSUE THREAD

**Issue by wazzamatazz** [DamianEdwards/MiniValidation#44]
When I try and validate an object that contains a `Uri` property MiniValidator throws a `TypeLoadException`:

```
Could not load type 'System.UriCreationOptions&' from assembly 'System.Private.Uri, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.

   at System.RuntimeTypeHandle.MakeByRef()
   at System.RuntimeType.MakeByRefType()
   at MiniValidation.TypeDetailsCache.<>c.<Visit>b__4_0(ParameterInfo p) in /_/src/MiniValidation/TypeDetailsCache.cs:line 61
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at MiniValidation.TypeDetailsCache.Visit(Type type, HashSet`1 visited, Boolean& requiresAsync) in /_/src/MiniValidation/TypeDetailsCache.cs:line 61
   at MiniValidation.TypeDetailsCache.Visit(Type type, HashSet`1 visited, Boolean& requiresAsync) in /_/src/MiniValidation/TypeDetailsCache.cs:line 100
   at MiniValidation.TypeDetailsCache.Visit(Type type) in /_/src/MiniValidation/TypeDetailsCache.cs:line 34
   at MiniValidation.TypeDetailsCache.Get(Type type) in /_/src/MiniValidation/TypeDetailsCache.cs:line 24
   at MiniValidation.MiniValidator.RequiresValidation(Type targetType, Boolean recurse) in /_/src/MiniValidation/MiniValidator.cs:line 44
   at MiniValidation.MiniValidator.TryValidate[TTarget](TTarget target, Boolean recurse, Boolean allowAsync, IDictionary`2& errors) in /_/src/MiniValidation/MiniValidator.cs:line 104
   at MiniValidation.MiniValidator.TryValidate[TTarget](TTarget target, IDictionary`2& errors) in /_/src/MiniValidation/MiniValidator.cs:line 64
   at UserQuery.Main(), line 6
```

Example code to reproduce

```csharp
static void Main() {
    var model = new Model() {
        BaseAddress = new Uri("https://some-org.com/")
    };

    if (!MiniValidator.TryValidate(model, out var errors)) {
        Console.WriteLine("fail");
    }
}

public class Model {

    [Required]
    public Uri BaseAddress { get; set; }
    
}
```

**Comment by DamianEdwards** [comment]
Thanks for the report, I'll take a look.

**Comment by DamianEdwards** [comment]
The fix for this is available in [0.7.3](https://github.com/DamianEdwards/MiniValidation/releases/tag/v0.7.3)
